### ISSUE TITLE
KIM adds a condition to RuntimeCRs which provide details about the applied auditlog configuration

### ISSUE THREAD

**Issue by tobiscr** [kyma-project/infrastructure-manager#363]
**Description**

To make the cluster creation process more transparent, the result of the auditlog configuration will always be tracked as addtional condition with the `RuntimeCR`. The condition will indicate whether an auditlog could be found. 

Examples (just for better understanding):

```
# No Auditlog instance found
  - lastTransitionTime: "2024-08-28T08:54:48Z"
    message: 'No auditlog instance found for Azure region west-europe: auditlog configuration failed.'
    reason: AuditlogMissingMappingErr
    status: "False"
    type: Provisioned

# Auditlog instance found (optional - success cases can be tracked as condition but it's not mandatory)
  - lastTransitionTime: "2024-08-28T08:54:48Z"
    message: 'Auditlog instance `....` found for Azure region west-europe. Auditlog configuration successfully added to Shoot spec.'
    reason:
    status: "true"
    type: Provisioned
```

AC:
- [ ] Clarify with the team if it#s meaningful to track also successfully configured clusters, or foucs only on the cases where the auditlog could not be configured
- [ ] At least for error cases, the auditlog configuration step creates a Condition in the Runtime CR which includes a notification that the auditlog could not be successfully configured

**Reasons**

Increase transparency of the auditlog configuration during cluster creation / update processes.

**Attachments**

**Relates to**
https://github.com/kyma-project/infrastructure-manager/issues/362

**Comment by Disper** [comment]
example of condition applied when mapping is missing
```
    - lastTransitionTime: "2024-09-09T12:28:18Z"
      message: 'Error during enabling Audit Logs on shoot: kim-md-al6: missing mapping
        for selected region in provider config'
      reason: AuditLogMissingRegionMappingErr
      status: "False"
      type: AuditlogConfigured
    state: Failed
```

**Comment by Disper** [comment]
Implemented in #367
